examples/hacker-news/app/Stories.re

```diff
@@ -0,0 +1,265 @@
+	      id
+       }
+       time,
+  descendants
+     }
+     ... on Poll @bsRecord {
+       title
+     }
+     ... on Job @bsRecord {
+       title
+     }
+   }
+  }
+}|}
+    ];
+
+    let toGraphQLClientQuery = (extractStories, query) => {
+      {
+        GraphQLClient.query: query#query,
+        variables: query#variables,
+        parse: json =>
+          Core_kernel.Option.map(query#parse(json)#hn, ~f=extractStories),
+      };
+    };
+
+    let make = (~storyType, ~offset, ~limit, ()) => {
+      switch (storyType) {
+      | Top =>
+        Top.make(~offset, ~limit, ())
+        |> toGraphQLClientQuery(x => x.topStories)
+      | Jobs =>
+        Jobs.make(~offset, ~limit, ())
+        |> toGraphQLClientQuery(x => x.jobStories)
+      | New =>
+        New.make(~offset, ~limit, ())
+        |> toGraphQLClientQuery(x => x.newStories)
+      | Show =>
+        Show.make(~offset, ~limit, ())
+        |> toGraphQLClientQuery(x => x.showStories)
+      };
+    };
+  };
+};
+
+let component = {
+  open Brisk_macos;
+  let component = Brisk.component("Stories");
+  (~children as _: list(unit), ~renderToolbar, ()) =>
+    component(hooks => {
+      let (selectedStoryType, setSelectedStoryType, hooks) =
+        Brisk.Hooks.state(StoryType.Top, hooks);
+      let (selectedStory, setSelectedStory, hooks) =
+        Brisk.Hooks.state(None, hooks);
+      let hooks =
+        Brisk.Hooks.effect(
+          Always,
+          () => {
+            open StoryType;
+            open Toolbar;
+            let storyItem = (~children as _: list(unit), ~storyType, ()) =>
+              SegmentedControl.(
+                <item
+                  isSelected={storyType === selectedStoryType}
+                  onClick={() => setSelectedStoryType(storyType)}>
+                  ...{toString(storyType)}
+                </item>
+              );
+            let segmentedControl = SegmentedControl.segmentedControl;
+            let () =
+              if (selectedStory === None) {
+                renderToolbar(
+                  <toolbar>
+                    <flexibleSpace />
+                    <item>
+                      <segmentedControl>
+                        <storyItem storyType=Top />
+                        <storyItem storyType=New />
+                        <storyItem storyType=Show />
+                        <storyItem storyType=Jobs />
+                      </segmentedControl>
+                    </item>
+                    <flexibleSpace />
+                  </toolbar>,
+                );
+              } else {
+                renderToolbar(
+                  <toolbar>
+                    <item>
+                      <button
+                        title="Open in Browser"
+                        bezel=Cocoa.BriskButton.BezelStyle.TexturedRounded
+                      />
+                    </item>
+                    <flexibleSpace />
+                    <item>
+                      <button
```

Comment:
This button is not square. We have to add an issue for it; it's a symptom of another problem with sizing things in the sidebar.

Reply:
You mean with toolbar? I had major issues working with NSToolbar, because of NSButton frames acting very odd.

Reply:
I did mean toolbar indeed. NSToolbar is a funny thing.